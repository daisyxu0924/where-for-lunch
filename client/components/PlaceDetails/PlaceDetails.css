.root {
  background: rgba(39, 38, 38, 0.67);
  padding: 20px 10px;
  border-radius: 15px;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
}

.header {
  & h1 > .badge {
    font-size: 12px;
    font-weight: normal;
    padding: 10px;
    margin-left: 10px;
    display: inline-block;
    border-radius: 6px;

    &.red {
      background: #ad0000;
    }

    &.green {
      background: #008000;
    }

    &.blue {
      background: #1c27ea;
    }
  }
}

.row {
  display: flex;
  flex-wrap: wrap;
}

.contactDetails {
  flex: 70%;

  & p > a {
    color: #ffeb3b;
    font-weight: bold;
  }
}

.photos {
  align-items: center;
  background: #000;
  flex: 30%;
  width: 200px;

  & img {
    width: 100%;
  }

  & > span {
    display: block;
    cursor: pointer;
    text-align: center;
  }
}