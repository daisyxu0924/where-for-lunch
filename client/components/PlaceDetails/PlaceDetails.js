import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Stars from 'components/Stars/Stars';
import Slider from 'react-slick';
import classNames from 'classnames/bind';
import styles from './PlaceDetails.css';

class PlaceDetails extends PureComponent {
  static propTypes = {
    place: PropTypes.object.isRequired,
  };

  render() {
    const { place } = this.props;
    const settings = {
      dots: true,
      infinite: true,
      speed: 500,
      lazyLoad: true,
      slidesToShow: 1,
      slidesToScroll: 1,
      pauseOnHover: true,
    };
    const isOpenData = (!place.is_closed) ?
      { text: 'OPEN', style: styles.green } :
      { text: 'CLOSED', style: styles.red };

    return (
      <section className={styles.root}>
        <header className={styles.header}>
          <h1>
            {place.name}
            <span className={classNames({
              [styles.badge]: true,
              [isOpenData.style]: true,
            })}>{isOpenData.text}</span>

            {
              place.is_claimed && (
                <span className={classNames({
                  [styles.badge]: true,
                  [styles.blue]: true,
                })}>CLAIMED</span>
              )
            }
          </h1>
          {
            place.rating && (
              <div>
                <span>{place.review_count} reviews</span>
                <Stars rating={place.rating}/>
              </div>
            )
          }
          <span>{place.price}</span>
          <p>
            {
              place.categories && place.categories.map(item => item.title).join(', ')
            }
          </p>

        </header>

        <div className={styles.row}>
          <div className={styles.contactDetails}>
            <p>
              Address:&nbsp;
              { place.location && place.location.display_address.join(', ') }
            </p>

            {
              place.display_phone && (
                <p>
                  Contact:&nbsp;
                  <a href={`tel:${place.phone}`}>
                    { place.display_phone }
                  </a>
                </p>
              )
            }
          </div>

          {
            place.photos && place.photos.length > 0 && (
              <div className={styles.photos}>
                <Slider {...settings}>
                  {place.photos.map(src => (
                    <div key={src}>
                      <img src={src} />
                    </div>
                  ))}
                </Slider>
              </div>
            )
          }

        </div>
      </section>
    );
  }
}

export default PlaceDetails;
