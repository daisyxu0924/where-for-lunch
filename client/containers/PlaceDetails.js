import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import placeActions from 'actions/placeActions';
import PlaceDetail from 'components/PlaceDetail/PlaceDetail';

class PlaceDetails extends Component {
  constructor(props) {
    super(props);

    if (this.props.match) {
      const id = this.props.match.params.id;

      this.props.fetchPlaceDetails(id);
    }
  }

  render() {
    const { details } = this.props.place;
    return (
      <div className="placeDetailsWrapper">
        { details ? <PlaceDetail details={details}/> : 'Loading, please wait'}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  place: state.place,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    fetchPlaceDetails: placeActions.fetchPlaceDetails,
  }, dispatch);

PlaceDetails.propTypes = {
  match: PropTypes.object,
  place: PropTypes.object,
  fetchPlaceDetails: PropTypes.func,
};
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PlaceDetails);
