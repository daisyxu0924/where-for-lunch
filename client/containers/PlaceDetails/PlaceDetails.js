import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import GoogleMap from 'components/GoogleMap/GoogleMap';
import placeActions from 'actions/placeActions';
import styles from './PlaceDetails.css';

class PlaceDetails extends Component {

  componentDidMount() {
    this.props.fetchPlaceDetails(this.props.match.params.id);
  }

  render() {
    const details = this.props.placeDetails;
    let images = null;
    if (details.photos) {
        images = details.photos.map(image => {
           return <img key={image} src={image} />
        }); 
    }

    return (
      <div>
        <div className={styles.root}>
          <div className={styles.box}>
            <div>{ details.address }</div>
            <div>{ details.phone }</div>
            <div>{ details.categories }</div>
            <div>{ details.price }</div>
            { details.coordinates ? <GoogleMap coordinates={details.coordinates} googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyC4R6AN7SmujjPUIGKdyao2Kqitzr1kiRg&v=3.exp&libraries=geometry,drawing,places" loadingElement={<div style={{ height: `100%` }} />} containerElement={<div style={{ height: `400px` }} />} mapElement={<div style={{ height: `100%` }} />} /> : null }
            {images}
          </div>
        </div>
      </div>
    );
  }
};

const mapStateToProps = state => ({
  placeDetails: state.placeDetails
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    fetchPlaceDetails: placeActions.fetchPlaceDetails,
  }, dispatch);

PlaceDetails.propTypes = {
  place: PropTypes.object,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PlaceDetails);