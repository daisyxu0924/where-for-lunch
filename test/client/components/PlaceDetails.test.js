import React from 'react';
import { shallow } from 'enzyme';
import PlaceDetails from 'client/components/PlaceDetails/PlaceDetails';
import renderer from 'react-test-renderer';

describe('PlaceDetails Component', () => {
  const props = {
    place: {
      rating: 5,
    },
  };

  test('renders Stars Component', () => {
    const wrapper = shallow(<PlaceDetails {...props} />);
    expect(wrapper.find('Stars').length).toBe(1);
  });

  test('renders properly', () => {
    const placeComponent = renderer
      .create(<PlaceDetails {...props} />)
      .toJSON();
    expect(placeComponent).toMatchSnapshot();
  });
});
