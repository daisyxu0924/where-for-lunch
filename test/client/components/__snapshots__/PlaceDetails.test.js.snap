// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceDetails Component renders properly 1`] = `
<section
  className="root"
>
  <header
    className="header"
  >
    <h1>
      <span
        className="badge green"
      >
        OPEN
      </span>
    </h1>
    <div>
      <span>
         reviews
      </span>
      <span
        className="root"
      >
        <div
          className="rating"
        >
          <div
            className="stars"
          >
            <div
              className="emptyStars"
            />
            <div
              className="fullStars"
              style={
                Object {
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
      </span>
    </div>
    <span />
    <p />
  </header>
  <div
    className="row"
  >
    <div
      className="contactDetails"
    >
      <p>
        Address:Â 
      </p>
    </div>
  </div>
</section>
`;
